tutorials-v5/heom/heom-3-quantum-heat-transport.md

```diff
@@ -0,0 +1,505 @@
+)
+ax1.plot(
+    tlist, np.real(result.expect[2]),
+    '--', color='darkorange', label='BHC (system -> bath 2)',
+)
+ax1.plot(
+    tlist, -np.real(result.expect[3]),
+    color='dodgerblue', label='SHC (bath 1 -> system)',
+)
+ax1.plot(
+    tlist, np.real(result.expect[4]),
+    '--', color='dodgerblue', label='SHC (system -> bath 2)',
+)
+
+ax1.set_xlabel('t', fontsize=28)
+ax1.set_ylabel('j', fontsize=28)
+ax1.set_ylim((-0.05, 0.05))
+ax1.legend(loc=0, fontsize=12)
+
+ax2.plot(
+    tlist, -np.real(result.expect[1]),
+    color='darkorange', label='BHC (bath 1 -> system)',
+)
+ax2.plot(
+    tlist, np.real(result.expect[2]),
+    '--', color='darkorange', label='BHC (system -> bath 2)',
+)
+ax2.plot(
+    tlist, -np.real(result.expect[3]),
+    color='dodgerblue', label='SHC (bath 1 -> system)',
+)
+ax2.plot(
+    tlist, np.real(result.expect[4]),
+    '--', color='dodgerblue', label='SHC (system -> bath 2)',
+)
+
+ax2.set_xlabel('t', fontsize=28)
+ax2.set_xlim((20, 50))
+ax2.set_ylim((0, 0.0002))
+ax2.legend(loc=0, fontsize=12);
+```
+
+### Steady-state currents
+
+Here, we try to reproduce the HEOM curves in Fig. 3(a) of Ref. \[1\] by varying the coupling strength and finding the steady state for each coupling strength.
+
+```{code-cell} ipython3
+def heat_currents(sys, bath_p1, bath_p2, Nk, NC, options):
+    """ Calculate the steady sate heat currents for the given system and
+        bath.
+    """
+    bath1 = bath_p1.bath(Nk, tag="bath 1")
+    Q1 = bath_p1.Q()
+
+    bath2 = bath_p2.bath(Nk, tag="bath 2")
+    Q2 = bath_p2.Q()
+
+    b1delta, b1term = bath1.terminator()
+    b2delta, b2term = bath2.terminator()
+
+    solver = HEOMSolver(
+        qt.liouvillian(sys.H()) + b1term + b2term,
+        [bath1, bath2],
+        max_depth=NC,
+        options=options
+    )
+
+    _, steady_ados = solver.steady_state()
+
+    return (
+        bath_heat_current('bath 1', steady_ados, sys.H(), Q1, b1delta),
+        bath_heat_current('bath 2', steady_ados, sys.H(), Q2, b2delta),
+        system_heat_current('bath 1', steady_ados, sys.H(), Q1, b1delta),
+        system_heat_current('bath 2', steady_ados, sys.H(), Q2, b2delta),
+    )
+```
+
+```{code-cell} ipython3
+# Define number of points to use for the plot
+plot_points = 10  # use 100 for a smoother curve
+
+# Range of relative coupling strengths
+# Chosen so that zb_max is maximum, centered around 1 on a log scale
+zb_max = 4  # use 20 to see more of the current curve
+zeta_bars = np.logspace(
+    -np.log(zb_max),
+    np.log(zb_max),
+    plot_points,
+    base=np.e,
+)
+
+# Setup a progress bar
+progress = IntProgress(min=0, max=(3 * plot_points))
+display(progress)
+
+
+def calculate_heat_current(J12, zb, Nk, progress=progress):
+    """ Calculate a single heat current and update the progress bar. """
+    # Estimate appropriate HEOM max_depth from coupling strength
+    NC = 7 + int(max(zb * J12 - 1, 0) * 2)
+    NC = min(NC, 15)
```

Comment:
I tried again with the full plot_points=100 and zb_max = 20, and  I  think the original notebook had NC=20 for the large combinations of values of J12 and Zb to get converged results?  With this truncation at 15 I see loss convergence above Zb =10.  Not super important for the results as presented, but might be worth modifying the this to min(NC,20).

Reply:
Changed back to 20. Just need to check it isn't too slow now.